flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpillingAdaptiveSpanningRecordDeserializer.java

```diff
@@ -550,6 +550,7 @@ private void addNextChunkFromMemorySegment(MemorySegment segment, int offset, in
 				}
 				else {
 					spillingChannel.close();
+					spillingChannel = null;
```

Comment:
extract those two lines to a method and reuse in `clear()`?

Reply:
It's not quite the same logic though...in `clear()`, any exceptions from the `close()` call are ignored. Seems a bit messy putting this into a single method to be "re-usable"...